contracts/0.4.24/nos/NodeOperatorsRegistry.sol

```diff
@@ -1,452 +1,673 @@
-// SPDX-FileCopyrightText: 2020 Lido <info@lido.fi>
-
+// SPDX-FileCopyrightText: 2023 Lido <info@lido.fi>
 // SPDX-License-Identifier: GPL-3.0
 
-/* See contracts/COMPILERS.md */
+// See contracts/COMPILERS.md
 pragma solidity 0.4.24;
 
-import "@aragon/os/contracts/apps/AragonApp.sol";
-import "@aragon/os/contracts/common/IsContract.sol";
-import "@aragon/os/contracts/lib/math/SafeMath.sol";
-import "@aragon/os/contracts/lib/math/SafeMath64.sol";
-import "solidity-bytes-utils/contracts/BytesLib.sol";
+import {AragonApp} from "@aragon/os/contracts/apps/AragonApp.sol";
+import {SafeMath} from "@aragon/os/contracts/lib/math/SafeMath.sol";
+import {SafeMath64} from "@aragon/os/contracts/lib/math/SafeMath64.sol";
+import {UnstructuredStorage} from "@aragon/os/contracts/common/UnstructuredStorage.sol";
+
+import {Math64} from "../lib/Math64.sol";
+import {BytesLib} from "../lib/BytesLib.sol";
+import {MemUtils} from "../../common/lib/MemUtils.sol";
+import {MinFirstAllocationStrategy} from "../../common/lib/MinFirstAllocationStrategy.sol";
+import {SigningKeysStats} from "../lib/SigningKeysStats.sol";
 
-import "../interfaces/INodeOperatorsRegistry.sol";
-import "../lib/MemUtils.sol";
 
+interface IStETH {
+    function sharesOf(address _account) external view returns (uint256);
+    function transferShares(address _recipient, uint256 _sharesAmount) external returns (uint256);
+}
+
+/// @dev This interface describes only tiny part of the full interface, which NodeOperatorsRegistry must implement
+///      See 0.8.9/interface/IStakingModule.sol for the full version.
+///      We don't inherit 0.8.9 IStakingModule due to the solidity version conflict.
+interface IStakingModule {
+    event ValidatorsKeysNonceChanged(uint256 validatorsKeysNonce);
```

Comment:
why do we need to declare this event in an interface here? maybe it would be better to declare it among other event in the contract body?

Reply:
For me it brings more explicitness and kind of underlines in code (not comments) the fact that although NodeOperatorsRegistry doesn't inherit "actual" IStakingModule it does implements.

Reply:
outdated, resolving the convo